docs/grammar/SolidityParser.g4

```diff
@@ -364,7 +365,14 @@ locals [boolean visibilitySet = false, boolean mutabilitySet = false]
 /**
  * The declaration of a single variable.
  */
-variableDeclaration: type=typeName location=dataLocation? name=identifier;
+variableDeclaration
+:
+	type=typeName
+	(
+		transientLocation=Transient name=identifier?
+		| location=dataLocation? name=identifier
+	);
+
```

Comment:
As @ekpyron pointed out, this is changing grammar for variable declarations in general.

Unless you changed approach to also accept `transient` for other variable declarations, this shouldn't be needed.

Reply:
The PR does parse all these cases - which is fine as long as we reject it in analysis for now - which the tests seem to show that the PR does - tests which would fail the parsing-grammar-comparison, if this wasn't here :-).